### ISSUE TITLE
Fatal queuedtracking:monitor issue

### ISSUE THREAD

**Issue by german-drulyk** [piwik/plugin-QueuedTracking#269]
**Error:**

    ERROR     [2024-10-21 13:39:35] 5424  Uncaught exception: Error: Call to undefined function Piwik\Plugins\QueuedTracking\Commands\readline_callback_handler_install() in matomo\plugins\QueuedTracking\Commands\Monitor.php:74

**Platform:**
Windows Server 2022
PHP x64 8.3.10
Matomo 5.1.1
Queued Tracking 5.1.0

**Issue:**
Windows PHP **DOES NOT** ship with `readline_callback_handler_install()`; contrary to the claim made at https://www.php.net/manual/en/readline.installation.php

**PHP Versions tested:**
7.1.0 x64
7.4.9 x64
8.1.9 x64
8.3.10 x64
8.3.12 x64

**Additional resources**
https://stackoverflow.com/a/71178783

**Comment by AltamashShaikh** [comment]
@german-drulyk Sorry to hear this, can you please check if changing this [Line](https://github.com/matomo-org/plugin-QueuedTracking/blob/5.x-dev/Commands/Monitor.php#L74..L75) in `QueuedTracking\Commands\Monitor.php` with below code solves the issue for you ??

```
if (function_exists('readline_callback_handler_install')) {
            readline_callback_handler_install('', function () {
            });
        }
```

**Comment by AltamashShaikh** [comment]
@haristku FYI

**Comment by german-drulyk** [comment]
@AltamashShaikh Thank you for the quick response.

Yes, using `function_exists()` does alleviate the symptom.

**Comment by AltamashShaikh** [comment]
@german-drulyk Does everything else works as expected ?

**Comment by german-drulyk** [comment]
@AltamashShaikh

I am seeing no further errors in my logs and the queue does seem to be processing.

However, the CMD window no longer auto-refreshes. I have to terminate and re-invoke `queuedtracking:monitor`. This is not a major issue to me but I just wanted to make you aware of what I am experiencing.

Please let me know if you would like additional testing of something specific.

Thanks!

**Comment by haristku** [comment]
@german-drulyk @AltamashShaikh 

i was worried that this might happen, since i didn't have a chance to try the "monitor" feature in a windows environment.
i'll try to find a work around to replace the `readline_callback_handler_install` function.

brb..

**Comment by AltamashShaikh** [comment]
@german-drulyk Can you check if this [PR](https://github.com/matomo-org/plugin-QueuedTracking/pull/271) solves the problem completely for you ?

**Comment by german-drulyk** [comment]
@AltamashShaikh The PR does seem to solve the `readline_callback_handler_install()` fatal error but I have not tested the dev code nor reviewed the code in its entirety.

The conditional check resulting in [line 267](https://github.com/haristku/plugin-QueuedTracking/blob/16353cc3c8105da9d6d69b844a5a693a5e13a5dd/Commands/Monitor.php#L267) seems susceptible to false positives.

Simplifying the function like this would seem sufficient:

    private function interactiveCapability()
    {
        return function_exists('readline_callback_handler_install');
    }

**Comment by AltamashShaikh** [comment]
@german-drulyk Can you replace `Commands/Monitor.php` with this file and check [once](https://github.com/haristku/plugin-QueuedTracking/blob/16353cc3c8105da9d6d69b844a5a693a5e13a5dd/Commands/Monitor.php) ?

**Comment by german-drulyk** [comment]
@AltamashShaikh I replaced `Monitor.php` with the dev version, re-ran the command, and have found zero new errors.

Additionally, the window does auto-refresh the queue counts now.

**Comment by AltamashShaikh** [comment]
@german-drulyk Thanks for testing this :+1: 
We have requested the change for `interactiveCapability` on PR and should merge it once done and release a new version accordingly :+1:

**Comment by AltamashShaikh** [comment]
@german-drulyk Released[ v5.1.1](https://github.com/matomo-org/plugin-QueuedTracking/releases/tag/5.1.1) of the plugin with a fix.
